docs/advanced/Working_with_git.rst

```diff
@@ -0,0 +1,151 @@
+|  - ``git rebase -i HEAD~NUM``    |  - Modifying to a `head <http://bit.ly/1As3Raw>`_   |
+|  - ``git rebase -i bbc643cd^``   |  - Modify to specified commit bbc643cd              |
+|  - ``git rebase --abort``        |  - Abort a rebase                                   |
+|  - ``git reflog``                |  - Tracks the changesets to the tip of branch       |
++----------------------------------+-----------------------------------------------------+
+
+Issues with pull Requests
+=========================
+
+Problems new contributors face
+-------------------------------
+
+1. **Understand** your `problem <https://sethrobertson.github.io/GitFixUm/fixup.how-can-i-switch-my-git-repository-to-a-particular-commit>`_ first.
+
+#. Learn to **go to a certain commit**.
+
+   * **Soln:** Each commits has a hash value(SHA-1) which is fixed. `Switch <http://stackoverflow.com/questions/4940054/how-can-i-switch-my-git-repository-to-a-particular-commit>`_ to a commit.
+
+#. If you need to **modify a single commit** as per requested by the maintainer in the pull request **revert the old commit with a new commit and push it** mentioning the changes you have made:
+
+   * **Soln:** Please **understand** `reset and revert <http://stackoverflow.com/questions/2530060/can-you-explain-what-git-reset-does-in-plain-english>`_ .The ideal solution is to `revert previous commit <http://stackoverflow.com/questions/4114095/revert-to-a-previous-git-commit>`_ ,edit it and ``push it`` for changes that are **not published** or ``force push`` it for changes that **are published**._.You can also `undo <http://stackoverflow.com/questions/927358/undo-the-last-git-commit>`_ a commit and ``force push`` the changes made.
+
+   * **Note:** Commits **do not** technically **change** when we force push the commit ,reset or modify them.But the **hash gets updated**, and the new hashed commit gets tagged to that branch.You can easily find you previous commit on github too just by adding ``/commit/<branch>`` to the repo address.It will show you the **remote git log**.See the hash get changed after rewriting the commit.
+
+#. Sometimes you have **two or more commit on your pull request**,which is usually not desired by maintainers.The solution is to do an interactive rebase and squash the previous **N** commits on the particular branch to which the pull request represents:
+
+   * **Soln:** Please **understand** `interactive rebase <https://help.github.com/articles/about-git-rebase/>`_.Checkout the branch the pull request represents,count the **number[N]** of commits you need to `squash <http://stackoverflow.com/questions/2563632/how-can-i-merge-two-commits-into-one>`_ from the pull requests or ``git log`` on that branch,than ``git rebase -i HEAD~N``.N (**number of commits before**) is usually 2 if you want to **squash 2 commits**.Change the pick to squash.Save the configuration.Than force push the new commit ``git push -f origin <branch>``.
```

Comment:
Missing spaces:
+ "/>`_.Checkout"
+ "represents,count"
+ "branch,than" (also "than" should be "then")
+ "HEAD~N``.N"
+ "commits**.Change"
+ "squash.Save" and "configuration.Than" -- I would change this line to
```
Change "pick" to "squash", save the configuration, and then force push the new commit using ``git push -f origin <branch>``
```